src/Recommendations/RuleModelRepository.php

```diff
@@ -35,6 +36,16 @@ public function map($item)
         );
     }
 
+    public function hydrateOne(RuleModel $ruleModel)
+    {
+        $item = $this->get($ruleModel);
+        $ruleModel->setType($item['type']);
+        $ruleModel->setRuleId($item['rule_id']);
+        $ruleModel->setPublished(new DateTimeImmutable($item['published']));
```

Comment:
This gives us a reliable RuleModel created from the user input and added to by the database.